src/common/k8s-api/kube-api.ts

```diff
@@ -89,13 +90,31 @@ export interface IKubeResourceList {
   }[];
 }
 
-export interface IKubeApiCluster {
+export interface ILocalKubeApiConfig {
   metadata: {
     uid: string;
   }
 }
 
-export function forCluster<T extends KubeObject>(cluster: IKubeApiCluster, kubeClass: KubeObjectConstructor<T>): KubeApi<T> {
+/**
+ * @deprecated
+ */
+export interface IKubeApiCluster extends ILocalKubeApiConfig {}
+
+export interface IRemoteKubeApiConfig {
+  cluster: {
+    server: string;
```

Comment:
How about naming this `serverAddress`?

Reply:
I was following kubeconfig naming convention here.